src/Fantomas.Tests/NumberOfItemsInfixOperatorExpressionTests.fs

```diff
@@ -0,0 +1,277 @@
+module Fantomas.Tests.NumberOfItemsInfixOperatorExpressionTests
+
+open NUnit.Framework
+open FsUnit
+open Fantomas.Tests.TestHelper
+open Fantomas.FormatConfig
+
+let config =
+    { config with
+          NewlineInfixOperatorExpressionMultilineFormatter = NumberOfItems }
+
+[<Test>]
+let ``simple infix operator expressions`` () =
+    formatSourceString false """
+let WebApp =
+    route "/ping" >=> authorized >=> text "pong"
+
+x
+|> aVeryLongFunctionNameThatGoesOnAndOnForeverAlmost
+
+x |> f |> g
+    """ config
+    |> prepend newline
+    |> should equal """
+let WebApp =
+    route "/ping" >=> authorized >=> text "pong"
+
+x |> aVeryLongFunctionNameThatGoesOnAndOnForeverAlmost
+
+x
+|> f
+|> g
+"""
+
+[<Test>]
+let ``multiple pipe boolean expression`` () =
+    formatSourceString false """let result = a && b |>  f |>  g |>   h
+"""  config
+    |> prepend newline
+    |> should equal """
+let result =
+    a
+    && b
+       |> f
+       |> g
+       |> h
+"""
+
+[<Test>]
+let ``mathematical expressions`` () =
+    formatSourceString false """
+let w = x + y + z
+
+let w = x - y - z
+    """ config
+    |> prepend newline
+    |> should equal """
+let w = x + y + z
+
+let w = x - y - z
+"""
+
+[<Test>]
+let ``equal sign operator should not move to next line`` () =
+    formatSourceString false """let result =
+(typ.GetInterface(typeof<System.Collections.IEnumerable>.FullName) = null)
+"""  config
+    |> prepend newline
+    |> should equal """
+let result =
+    (typ.GetInterface(typeof<System.Collections.IEnumerable>.FullName) = null)
+"""
+
+[<Test>]
+let ``should not add newline before = operator after |>`` () =
+    formatSourceString false """1 |> max 0 = 1""" config
+    |> should equal """1 |> max 0 = 1
+"""
+
+[<Test>]
+let ``should add space around .. .. operators and not add newline`` () =
+    formatSourceString false """[10 .. -1 .. 1]""" config
+    |> should equal """[ 10 .. -1 .. 1 ]
+"""
+
+[<Test>]
+let ``long expression with pipe should be single line`` () =
+    formatSourceString false "let a = List.init 40 (fun i -> generateThing i a) |> List.map mapThingToOtherThing" config
+    |> prepend newline
+    |> should equal """
+let a =
+    List.init 40 (fun i -> generateThing i a) |> List.map mapThingToOtherThing
+"""
+
+[<Test>]
+let ``modulo operator on same line, 780`` () =
+    formatSourceString false """let hasUnEvenAmount regex line = (Regex.Matches(line, regex).Count - Regex.Matches(line, "\\\\" + regex).Count) % 2 = 1
+"""  config
+    |> prepend newline
+    |> should equal """
+let hasUnEvenAmount regex line =
+    (Regex.Matches(line, regex).Count - Regex.Matches(line, "\\\\" + regex).Count) % 2 = 1
```

Comment:
`(Regex.Matches(line, regex).Count - Regex.Matches(line, "\\\\" + regex).Count)` should be split here I think.